src/Message/Headers.php

```diff
@@ -14,7 +14,11 @@ public function __construct(\stdClass $headers)
         $headers = \array_change_key_case((array) $headers);
 
         foreach ($headers as $key => $value) {
-            $this[$key] = $this->parseHeader($key, $value);
+            try {
+                $this[$key] = $this->parseHeader($key, $value);
+            } catch (\Exception $e) {
```

Comment:
```suggestion
            } catch (\Ddeboer\Imap\Exception\UnsupportedCharsetException $e) {
```
Let just catch the only known types of Exception